sdk/storage/Azure.Storage.Common/src/Shared/StorageClientOptions.cs

```diff
@@ -113,7 +115,7 @@ public static HttpPipeline Build(this ClientOptions options, HttpPipelinePolicy
 
             return HttpPipelineBuilder.Build(
                options,
-               Array.Empty<HttpPipelinePolicy>(),
+               PerCallPolicies,
```

Comment:
Why per call?  Why not per retry?  See the other comment about decreasing this value for each retry.

Reply:
The resolution of this comment depends on closure on the other comment.
The challenging part will be that HttpMessage is shared by retries and timeout is part of query (see [here](https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations)), so some parsing/replacement would be involved if this is to be computed dynamically.

Reply:
>See the other comment about decreasing this value for each retry.

This feature is specifically about how long the call takes on the service. I don't think we should reduce the timeout just because the first two calls timed out before reaching the service.